src/dali/plugin/input/rest/binance_com.py

```diff
@@ -0,0 +1,774 @@
+            # If the user converts more than 100 assets at one time, we can not retrieve accurate records.
+            if len(dust_trades) == 100:
+                current_dribblet: Any = []
+                current_dribblet_time: int = dust_trades[0][_DIVTIME]
+                for dust in dust_trades:
+                    dust[_ID] = dust[_ORDER]
+                    if dust[_DIVTIME] == current_dribblet_time:
+                        current_dribblet.append(dust)
+                    elif len(current_dribblet) < 101:
+                        for dribblet_piece in current_dribblet:
+                            self._process_sell(dribblet_piece, out_transactions)
+                            self._process_buy(dribblet_piece, in_transactions, out_transactions)
+
+                        # Shift the call window forward past this dribblet
+                        current_start = current_dribblet_time + 1
+                        current_end = current_start + 2592000000
+                        break
+                    else:
+                        raise Exception(f"Too many assets dusted at the same time: " f"{self._rp2timestamp_from_ms_epoch(current_dribblet_time)}")
+            else:
+
+                for dust in dust_trades:
+                    # dust trades have a null id
+                    dust[_ID] = dust[_ORDER]
+                    self._process_sell(dust, out_transactions)
+                    self._process_buy(dust, in_transactions, out_transactions)
+
+                current_start = current_end + 1
+                current_end = current_start + 2592000000
+
+    ### Single Transaction Processing
+
+    def _process_buy(
+        self, transaction: Any, in_transaction_list: List[InTransaction], out_transaction_list: List[OutTransaction], notes: Optional[str] = None
+    ) -> None:
+        self.__logger.debug("Buy Transaction: %s", transaction)
+        unique_id: str
+        timestamp: str
+        spot_price: RP2Decimal
+        crypto_in: RP2Decimal
+        crypto_fee: RP2Decimal
+
+        if _ISFIATPAYMENT in transaction:
+            unique_id = transaction[_ORDERNO]
+            timestamp = self._rp2timestamp_from_ms_epoch(transaction[_CREATETIME])
+            in_asset = transaction[_CRYPTOCURRENCY]
+            spot_price = RP2Decimal(transaction[_SOURCEAMOUNT]) / RP2Decimal(transaction[_OBTAINAMOUNT])
+            crypto_in = transaction[_OBTAINAMOUNT]
+            crypto_fee = None
+            fiat_in_no_fee = str(transaction[_SOURCEAMOUNT])
+            fiat_in_with_fee = str(RP2Decimal(transaction[_SOURCEAMOUNT]) - RP2Decimal(transaction[_TOTALFEE]))
+            fiat_fee = str(RP2Decimal(transaction[_TOTALFEE]))
+            transaction_notes = f"Buy transaction for fiat payment orderNo - " f"{transaction[_ORDERNO]}"
+        else:
+            trade = self._to_trade(transaction[_SYMBOL], str(transaction[_AMOUNT]), str(transaction[_COST]))

```

Comment:
Is this else a conversion? If so, maybe add a comment stating so.

Reply:
This is a fiat buy (e.g. a credit card payment in USD for a crypto asset).

Binance allows users to buy assets by credit card in pretty much any major currency, but they only trade in around 6-7 mostly European currencies.